### PR TITLE
Implement Debug and PartialEq for WinRT types

### LINKED ISSUE #104: Support for procedural Xaml
While you can import `windows.ui.xaml`, importing `windows.ui.xaml.controls` fails for a few different reasons. I've resolved a few of them, but at least one more remains (structs with `IReference<T>` members). Coming soon.

* WinRT types (identified by `RuntimeType`) must implement the `Debug` trait. This will allow `IReference<T>` to implement the `Debug` trait that will then allow WinRT structs that already implement this trait to compile.
* WinRT types must also implement the `PartialEq` trait. This again supports structs with `IReference<T>` fields. For class/interface types this should query for `IUnknown` and compare those pointer values for COM identity.

### PR DESCRIPTION
**Pull request by kennykerr**
Fixes #104 #131 #134

### CODE REVIEW

File: crates/winmd/src/types/delegate.rs [hunk 1]
```diff
@@ -105,6 +105,15 @@ impl Delegate {
                     self.ptr.set_abi()
                 }
             }
+            impl<#constraints> ::std::fmt::Debug for #name {
+                fn fmt(&self, f: &mut ::std::fmt::Formatter<'_>) -> ::std::fmt::Result {
+                    write!(
+                        f,
+                        "{:?}",
```
Comment (rylev) [inline crates/winmd/src/types/delegate.rs:112]: This doesn't seem too helpful. Shouldn't we also print the name of the type as well?

File: crates/winmd/src/types/delegate.rs [hunk 2]
```diff
@@ -60,7 +60,7 @@ impl Delegate {
 
         quote! {
             #[repr(transparent)]
-            #[derive(Default)]
+            #[derive(Default, PartialEq)]
```
Comment (rylev) [inline crates/winmd/src/types/delegate.rs:63]: Derives do the conservative thing and place a matching constraint on all generics for the derived trait. So for example:
```rust
#[derive(PartialEq)]
struct Foo<T> {
  f: std::marker::PhantomData<T>
}
// Foo<T> is only PartialEq when T: PartialEq
```
Is this what we want? Or do we just want to differ to the `self.ptr == other.ptr`? If so, we need to implement `PartialEq` ourselves.

Comment (kennykerr) [reply crates/winmd/src/types/delegate.rs:63]: Ah right. Yes, we just want to compare the pointers.

File: crates/winmd/src/types/delegate.rs (hunk 1 above)
Comment (kennykerr) [reply crates/winmd/src/types/delegate.rs:112]: The was primarily for completeness to unblock the issues with importing WinRT modules that used WinRT structs that have interface fields. WinRT structs mainly have primitive value fields and its very useful to have the Debug trait for those, but there are a handful that also have interface fields. I created https://github.com/microsoft/winrt-rs/issues/138 to address making these more helpful.

File: crates/winmd/src/types/delegate.rs (hunk 2 above)
Comment (kennykerr) [reply crates/winmd/src/types/delegate.rs:63]: https://github.com/microsoft/winrt-rs/issues/140

**Pull request merged**